src/finch-mcp-server/awslabs/finch_mcp_server/utils/build.py

```diff
@@ -0,0 +1,141 @@
+
+    Args:
+        dockerfile_path (str): Path to the Dockerfile to check.
+
+    Returns:
+        bool: True if the Dockerfile contains ECR references, False otherwise.
+
+    """
+    try:
+        if not os.path.exists(dockerfile_path):
+            logger.warning(f'Dockerfile not found at {dockerfile_path}')
+            return False
+
+        with open(dockerfile_path, 'r') as f:
+            content = f.read()
+            return bool(re.search(ECR_REFERENCE_PATTERN, content))
+    except Exception as e:
+        logger.error(f'Error checking Dockerfile for ECR references: {str(e)}')
+        return False
+
+
+def build_image(
+    dockerfile_path: str,
+    context_path: str,
+    tags: Optional[List[str]] = None,
+    platforms: Optional[List[str]] = None,
+    target: Optional[str] = None,
+    no_cache: Optional[bool] = False,
+    pull: Optional[bool] = False,
+    build_contexts: Optional[List[str]] = None,
+    outputs: Optional[str] = None,
+    cache_from: Optional[List[str]] = None,
+    quiet: Optional[bool] = False,
+    progress: Optional[str] = None,
+) -> Dict[str, Any]:
+    """Build a container image using Finch.
+
+    Args:
+        dockerfile_path: Path to the Dockerfile
+        context_path: Path to the build context directory
+        tags: List of tags to apply to the image
+        platforms: List of target platforms
+        target: Target build stage
+        no_cache: Whether to disable cache
+        pull: Whether to always pull base images
+        build_contexts: List of additional build contexts
+        outputs: Output destination
+        cache_from: List of external cache sources
+        quiet: Whether to suppress build output
+        progress: Type of progress output
+    Returns:
+        Dict[str, Any]: Result of the build operation
+
+    """
+    try:
+        # Check if Dockerfile exists
+        if not os.path.exists(dockerfile_path):
+            return format_result(STATUS_ERROR, f'Dockerfile not found at {dockerfile_path}')
+
+        if not os.path.exists(context_path):
+            return format_result(STATUS_ERROR, f'Context directory not found at {context_path}')
+
+        command = ['finch', 'image', 'build']
+
+        command.extend(['-f', dockerfile_path])
+
+        if tags:
+            for tag in tags:
+                command.extend(['-t', tag])
+
+        if platforms:
+            for platform in platforms:
+                command.extend(['--platform', platform])
+
+        if target:
+            command.extend(['--target', target])
+
+        if no_cache:
+            command.append('--no-cache')
+
+        if pull:
+            command.append('--pull')
+
+        if build_contexts:
+            for ctx in build_contexts:
+                command.extend(['--build-context', ctx])
+
+        if outputs:
+            command.extend(['--output', outputs])
+
+        if cache_from:
+            for cache in cache_from:
+                command.extend(['--cache-from', cache])
```

Comment:
Do we need to do any deduping for the arguments which take lists (`tags`, `platforms`, `build-context`), or any rules for which of these flags can be used together? Similar comments apply for other commands